mesonbuild/compilers/detect.py

```diff
@@ -171,7 +171,14 @@ def detect_static_linker(env: 'Environment', compiler: Compiler) -> StaticLinker
             trials = [defaults['gcc_static_linker']] + default_linkers
         elif compiler.id == 'clang':
             # Use llvm-ar if available; needed for LTO
-            trials = [defaults['clang_static_linker']] + default_linkers
+            llvm_ar = defaults['clang_static_linker']
+            extra_linkers = []
+            exe = compiler.name_string()
+            suffix = exe.rsplit('-')[-1]
+            if suffix != exe:
+                extra_linkers.append([f'{llvm_ar[0]}-{suffix}'])
+            extra_linkers.append(llvm_ar)
+            trials = extra_linkers + default_linkers
```

Comment:
Once the string for the suffixed llvm-ar is defined, I think you can just do "if suffixed_llvm_ar: trials = [suffixed_llvm_ar] + trials".